src/main/java/uk/gov/hmcts/reform/roleassignment/controller/endpoints/GetAssignmentController.java

```diff
@@ -164,4 +163,41 @@ public Case getCaseDataV2(@PathVariable("caseId") String caseId) {
         }
         return ResponseEntity.status(HttpStatus.OK).body(rootNode);
     }
+
+    @GetMapping(
+        path = "/am/role-assignments",
+        produces = V1.MediaType.GET_ASSIGNMENTS
+    )
+    @ApiOperation("Get Role assignment records by Case Id and Actor Id for RoleType as a CASE.")
+    @ApiResponses({
+                      @ApiResponse(
+                          code = 200,
+                          message = "Success",
+                          response = RoleAssignmentRequestResource.class
+                      ),
+                      @ApiResponse(
+                          code = 400,
+                          message = V1.Error.INVALID_REQUEST
+                      ),
+                      @ApiResponse(
+                          code = 404,
+                          message = V1.Error.NO_RECORDS_FOUND_BY_ACTOR
+                      ),
+                      @ApiResponse(
+                          code = 404,
+                          message = V1.Error.NO_RECORDS_FOUND_FOR_CASE_ID
+                      )
+                  })
+    public ResponseEntity<Object> retrieveRoleAssignmentsByActorIdAndCaseId(
+        @ApiParam(value = "Actor Id", required = false)
+        @RequestParam(value = "actorId", required = false) String actorId,
+        @ApiParam(value = "Case Id", required = false)
+        @RequestParam(value = "caseId", required = false) String caseId,
+        @ApiParam(value = "Role Type", required = true)
+        @RequestParam(value = "roleType", required = true) String roleType) {
+        if (StringUtils.isEmpty(roleType) || !roleType.equals(RoleType.CASE.name())) {
+            throw new BadRequestException(V1.Error.INVALID_ROLE_TYPE);
```

Comment:
please move the exception handling at Orchestrator or Parse service.

Reply:
Refactored the code to a new controller and parse service.